src/cpp/session/modules/NotebookData.R

```diff
@@ -19,7 +19,20 @@
     "print.data.frame" = function(x, options) list(x = x, options = options, className = class(x), nRow = .rs.scalar(nrow(x)), nCol = .rs.scalar(ncol(x))),
     "print.tbl_df" = function(x, options)     list(x = x, options = options, className = class(x), nRow = .rs.scalar(nrow(x)), nCol = .rs.scalar(ncol(x))),
     "print.grouped_df" = function(x, options) list(x = x, options = options, className = class(x), nRow = .rs.scalar(nrow(x)), nCol = .rs.scalar(ncol(x))),
-    "print.data.table" = function(x, options) list(x = x, options = options, className = class(x), nRow = .rs.scalar(nrow(x)), nCol = .rs.scalar(ncol(x))),
+    "print.data.table" = function(x, options) {
+      shouldPrintTable <- TRUE
+      if (exists(x = "shouldPrint", envir = as.environment("package:data.table"))) {
+        shouldPrintTable <- data.table::shouldPrint(x)
+      } else {
+        shouldPrintTable <- data.table:::.global$print == "" || data.table::address(x) != data.table:::.global$print
```

Comment:
We should avoid accessing `data.table` internal APIs (or if we do we need to wrap it in a `tryCatch` since there's no guarantee that will stay the same across versions of `data.table`).